quickjs.c

```diff
@@ -47678,6 +47786,20 @@ static JSValue js_typed_array_get_byteLength(JSContext *ctx,
         }
     }
     ta = p->u.typed_array;
+    if (typed_array_is_variable_length(ctx, p)) {
+        JSArrayBuffer *abuf = ta->buffer->u.array_buffer;
+        if (ta->is_length_tracking) {
+            /* Check if RAB has shrunk & start offset is OOB */
+            if (ta->offset > abuf->byte_length)
+                return JS_ThrowRangeError(ctx, "invalid typed array offset");
+            return JS_NewInt32(ctx, (abuf->byte_length - ta->offset));
```

Comment:
```suggestion
            return JS_NewUint32(ctx, abuf->byte_length - ta->offset);
```
Because `ta->offset` is `uint32_t`. `abuf->byte_length` probably should be too but one thing at a time.